### PR TITLE
Stop using cloudFoundryCloudProvider.id

### PR DESCRIPTION
**Pull request by gregturn**

### CODE REVIEW

File: clouddriver-cf/src/main/groovy/com/netflix/spinnaker/clouddriver/cf/provider/agent/ClusterCachingAgent.groovy [hunk 1]
```diff
@@ -86,7 +86,7 @@ class ClusterCachingAgent implements CachingAgent, OnDemandAgent, AccountAware {
     this.cloudFoundryClientFactory = cloudFoundryClientFactory
     this.cloudFoundryCloudProvider = cloudFoundryCloudProvider
```
Comment (cfieber) [inline clouddriver-cf/src/main/groovy/com/netflix/spinnaker/clouddriver/cf/provider/agent/ClusterCachingAgent.groovy]: doesn't look like this needs to be passed in here anymore

**Comment by gregturn** [comment]
@cfieber Based on your feedback, I spotted a couple other places that no longer needed it. Updated the unit test.

**Comment by gregturn** [comment]
@cfieber please review.

**Comment by cfieber** [comment]
LGTM

**Comment by gregturn** [comment]
@cfieber rebased.

**Pull request merged**